service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -157,15 +155,19 @@ private void removeDocumentFromData(Map<String, Object> data) {
     private List<Element<C2DocumentBundle>> buildC2DocumentBundle(CaseData caseData) {
         List<Element<C2DocumentBundle>> c2DocumentBundle = defaultIfNull(caseData.getC2DocumentBundle(),
             Lists.newArrayList());
+        String uploadedBy = documentUploadHelper.getUploadedDocumentUserDetails();
 
         List<SupportingEvidenceBundle> updatedSupportingEvidenceBundle =
             unwrapElements(caseData.getTemporaryC2Document().getSupportingEvidenceBundle())
                 .stream()
-                .map(supportingEvidence -> supportingEvidence.toBuilder().dateTimeUploaded(time.now()).build())
+                .map(supportingEvidence -> supportingEvidence.toBuilder()
+                    .dateTimeUploaded(time.now())
+                    .uploadedBy(uploadedBy)
+                    .build())
                 .collect(Collectors.toList());
 
         var c2DocumentBundleBuilder = caseData.getTemporaryC2Document().toBuilder()
-            .author(idamClient.getUserInfo(requestData.authorisation()).getName())
+            .author(uploadedBy)
```

Comment:
Do we want to update author here? I thought this would effect just supportingEvidenceBundle no?

Reply:
6th A/C says c2 upload event and I asked this question in some earlier meeting. I was asked to change 🤷‍♂️

Reply:
Maybe best to show Gail and explain this was existing functionality, do we need to change existing functionality.

Reply:
I have attached screenshot to story asking Gail and Eilis and asked Jay in a meeting and they all said yes but I will ask again.

Reply:
Gail confirmed saying this is correct.